File: homework2/app/src/main/java/com/example/homework2/MainActivity.java
package com.example.homework2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    // 宣告全域變數
    private TextView txtShow;
    private Button btnZero, btnOne, btnTwo, btnThree, btnFour;
    private Button btnFive, btnSix, btnSeven, btnEight, btnNine, btnClear;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // 取得資源類別檔中的介面元件
        txtShow = (TextView) findViewById(R.id.txtShow);
        btnZero = (Button) findViewById(R.id.btnZero);
        btnOne = (Button) findViewById(R.id.btnOne);
        btnTwo = (Button) findViewById(R.id.btnTwo);
        btnThree = (Button) findViewById(R.id.btnThree);
        btnFour = (Button) findViewById(R.id.btnFour);
        btnFive = (Button) findViewById(R.id.btnFive);
        btnSix = (Button) findViewById(R.id.btnSix);
        btnSeven = (Button) findViewById(R.id.btnSeven);
        btnEight = (Button) findViewById(R.id.btnEight);
        btnNine = (Button) findViewById(R.id.btnNine);
        btnClear = (Button) findViewById(R.id.btnClear);

        // 設定 button 元件 Click 事件共用   myListner
        btnZero.setOnClickListener(myListner);
        btnOne.setOnClickListener(myListner);
        btnTwo.setOnClickListener(myListner);
        btnThree.setOnClickListener(myListner);
        btnFour.setOnClickListener(myListner);
        btnFive.setOnClickListener(myListner);
        btnSix.setOnClickListener(myListner);
        btnSeven.setOnClickListener(myListner);
        btnEight.setOnClickListener(myListner);
        btnNine.setOnClickListener(myListner);
        btnClear.setOnClickListener(myListner);
    }

    // 定義  onClick() 方法
    public View.OnClickListener myListner = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            String s = txtShow.getText().toString();
            int id = v.getId(); // 先取得 ID

            if (id == R.id.btnZero) {
                txtShow.setText(s + "0");
            } else if (id == R.id.btnOne) {
                txtShow.setText(s + "1");
            } else if (id == R.id.btnTwo) {
                txtShow.setText(s + "2");
            } else if (id == R.id.btnThree) {
                txtShow.setText(s + "3");
            } else if (id == R.id.btnFour) {
                txtShow.setText(s + "4");
            } else if (id == R.id.btnFive) {
                txtShow.setText(s + "5");
            } else if (id == R.id.btnSix) {
                txtShow.setText(s + "6");
            } else if (id == R.id.btnSeven) {
                txtShow.setText(s + "7");
            } else if (id == R.id.btnEight) {
                txtShow.setText(s + "8");
            } else if (id == R.id.btnNine) {
                txtShow.setText(s + "9");
            } else if (id == R.id.btnClear) {
                txtShow.setText("電話號碼：");
            }
        }

    };
}

File: orderMenu/app/src/main/java/com/example/ordermenu/MainActivity.kt
package com.example.ordermenu

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.PixelCopy
import android.widget.Button
import android.widget.CheckBox
import android.widget.RadioButton
import androidx.appcompat.app.AppCompatActivity
import android.widget.TextView
import androidx.activity.result.ActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import androidx.core.os.bundleOf

class MainActivity : AppCompatActivity() {

    private lateinit var btnMainMeal: Button
    private lateinit var btnSideDish: Button
    private lateinit var btnDrink: Button
    private lateinit var btnComfirm: Button

    private lateinit var tvMeal: TextView
    private lateinit var tvSide: TextView
    private lateinit var tvDrink: TextView



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViews()
        setListeners()
        tvMeal=findViewById(R.id.txtM)
        tvDrink=findViewById(R.id.txtD)
        tvSide=findViewById(R.id.txtS)
    }

    private val MainMealResult = registerForActivityResult(
        ActivityResultContracts.StartActivityForResult()
    )
    {
            result: ActivityResult->
        if (result.resultCode==Activity.RESULT_OK){
            var intent=result.data
            var MainMeal=intent?.getStringExtra("GPMainMeal")
            tvMeal.text="Main-$MainMeal"
        }
    }

    private val SideDishResult = registerForActivityResult(
        ActivityResultContracts.StartActivityForResult()
    )
    {
            result: ActivityResult->
        if (result.resultCode==Activity.RESULT_OK){
            var intent=result.data
            val Fries=intent?.getBooleanExtra("checkFries",false)?:false
            val Salad=intent?.getBooleanExtra("checkSalad",false)?:false
            val CornCup=intent?.getBooleanExtra("checkCornCup",false)?:false
            val txt= StringBuilder("SideDishes-")
            if(Fries){
                txt.append("Fries ")
            }
            if(Salad){
                txt.append("Salad ")

            }
            if(CornCup){
                txt.append("CornCup")
            }
            tvSide.text=txt.toString()
        }
        }

    private val DrinkResult = registerForActivityResult(
        ActivityResultContracts.StartActivityForResult()
    )
    {
            result: ActivityResult->
        if (result.resultCode==Activity.RESULT_OK){
            var intent=result.data
            var Drink=intent?.getStringExtra("GPDrink")
            tvDrink.text="Drink-$Drink"
        }
    }


    private fun findViews() {
        btnMainMeal = findViewById(R.id.maimmeal)
        btnSideDish = findViewById(R.id.sidedish)
        btnDrink = findViewById(R.id.drink)
        btnComfirm = findViewById(R.id.Comfim)
    }

    private fun setListeners() {
        btnMainMeal.setOnClickListener {
            val intent = Intent(this, MainMealActivity::class.java)
            MainMealResult.launch(intent)
        }

        btnSideDish.setOnClickListener {
            val intent = Intent(this, SideDishesActivity::class.java)
            SideDishResult.launch(intent)
        }

        btnDrink.setOnClickListener {
            val intent = Intent(this, DrinkActivity::class.java)
            DrinkResult.launch(intent)
        }
        btnComfirm.setOnClickListener {
            val intent = Intent(this, ComfirmActivity::class.java)
            val TXTM=tvMeal.text.toString()
            val TXTS=tvSide.text.toString()
            val TXTD=tvDrink.text.toString()
            intent.putExtra("OM", TXTM)
            intent.putExtra("OS", TXTS)
            intent.putExtra("OD", TXTD)
            startActivity(intent)
        }
    }
}




